app/controllers/provider_foreman_controller.rb

```diff
@@ -721,14 +750,20 @@ def configuration_profile_record?(node = x_node)
     type && ["ConfigurationProfile"].include?(TreeBuilder.get_model_for_prefix(type))
   end
 
+  def inventory_group_record?(node = x_node)
+    type, _id = node.split("_").last.split("-")
+    type && ["EmsFolder"].include?(TreeBuilder.get_model_for_prefix(type))
+  end
+
   def foreman_provider_record?(node = x_node)
     node = node.split("-").last if node.split("-").first == 'xx'
     type, _id = node.split("-")
     type && ["ManageIQ::Providers::Foreman::ConfigurationManager"].include?(TreeBuilder.get_model_for_prefix(type))
   end
 
   def ansible_tower_provider_record?(node = x_node)
-    node = node.split("-").last if node.split("-").first == 'xx'
+    return  @record.kind_of?(ManageIQ::Providers::AnsibleTower::ConfigurationManager) if @record
```

Comment:
This method name is misleading... return true id you have a Manager (ExtManagementSystem), not a Provider